trickster/model.py

```diff
@@ -0,0 +1,475 @@
+            min_delay = float(data[0])
+            max_delay = float(data[1])
+        else:
+            raise ValueError('Input of response delay must be a single value or list of two numbers.')
+
+        if min_delay > max_delay:
+            raise ValueError(f'Maximum delay ({max_delay}s) must be greater than minimum delay ({min_delay}s).')
+
+        return {
+            'min_delay': min_delay,
+            'max_delay': max_delay
+        }
+
+    def delay_response(self) -> None:
+        """Pause the program for specified time."""
+        time.sleep(random.uniform(self.min_delay, self.max_delay))
+
+
+class Response(BaseModel):
+    """User-defined response Trickster should return when a request matches a response."""
+
+    id: uuid.UUID = Field(default_factory=uuid.uuid4, description='Unique identifier')  # noqa: A003
+    hits: HitCounter
+
+    status_code: http.HTTPStatus = Field(description='Status code of the response as int')
+    body: JsonBody
+    delay: ResponseDelay = ResponseDelay()
+    headers: dict[str, str] = Field(default_factory=dict, description='Header of the response')
+    weight: float = Field(ge=0.0, default=1.0, description='Weight of the response when selecting random response')
+
+    def as_fastapi_response(self) -> JSONResponse:
+        """Create a response that can be returned by FastApi."""
+        return JSONResponse(
+            content=self.body,
+            status_code=self.status_code,
+            headers=self.headers
+        )
+
+    def delay_response(self) -> None:
+        """Stop program for a specified amount of time."""
+        self.delay.delay_response()
+
+
+class ResponseSelector(enum.Enum):
+    """Algorithm to select response from matched route.
+
+    - `RANDOM`: Select response by random from all available
+    - `FIRST`: Select first response from all available
+    - `BALANCED`: Select response that was returned the least
+    """
+
+    RANDOM = 'random'
+    FIRST = 'first'
+    BALANCED = 'balanced'
+
+    def select_response(self, responses: list[Response]) -> Response:
+        """Select response based on response selector."""
+        if not responses:
+            raise ValueError('No suitable response found.')
+
+        match self:
+            case ResponseSelector.FIRST:
+                return responses[0]
+            case ResponseSelector.RANDOM:
+                weights = [response.weight for response in responses]
+                return random.choices(responses, weights=weights, k=1)[0]
+            case ResponseSelector.BALANCED:
+                return min(responses, key=lambda response: response.hits)
+            case _:
+                raise ValueError(f'Response selection algorithm for {self.value} is not configured.')
+
+
+class Auth(BaseModel):
+    """Base class for authentication."""
+
+    type: str
+    error_response: Response
+
+    model_config = ConfigDict(extra='allow')
+
+    def authenticate(self, request: Request) -> None:
```

Comment:
this
`class Auth(BaseModel, abc.ABC):
    """Base class for authentication."""

    type: str
    error_response: Response

    model_config = ConfigDict(extra='allow')

    @abc.abstractmethod
    def authenticate(self, request: Request) -> None:
        """Implement authenticate method in subclass."""`
        
is not possible because route has Auth field and it will results in
TypeError: Can't instantiate abstract class Auth with abstract method authenticate